src/fmeval/data_loaders/data_sources.py

```diff
@@ -110,14 +114,25 @@ def __reduce__(self):
 
 def get_s3_client(uri: str) -> boto3.client:
     """
-    Util method to return boto3 s3 client. For built-in datasets, the boto3 client region is default to us-west-2 as
-        the bucket is not accessible in opt-in regions.
+    Util method to return boto3 s3 client. For built-in datasets, the boto3 client region is default to us-west-2 for
+        commercial regions as the bucket is not accessible in opt-in regions.
+        For us-isof partition, built-in datasets are located in us-isof-south-1 region.
+
     :param uri: s3 dataset uri
     :return: boto3 s3 client
     """
-    s3_client = (
-        boto3.client("s3", region_name=BUILT_IN_DATASET_DEFAULT_REGION)
-        if uri.startswith(BUILT_IN_DATASET_PREFIX)
-        else boto3.client("s3")
-    )
+    session = boto3.session.Session()
+    region = session.region_name
+    if region in BUILT_IN_DATASET_ISO_REGIONS.keys():
+        s3_client = (
+            boto3.client("s3", region_name=BUILT_IN_DATASET_ISO_REGIONS[region], verify=False)
+            if uri.startswith(BUILT_IN_DATASET_PREFIX)
+            else boto3.client("s3", verify=False)
```

Comment:
Non-blocking nit: Should we specify `region_name` when calling `boto3.client` just to be safe/make this detail more explicit? 

Marking this as a nit since based on the [docs](https://boto3.amazonaws.com/v1/documentation/api/1.13.23/guide/configuration.html), whatever configuration files/env vars are used to create `session` should be used to create the boto3 client as well, so the current code should work.